aiosparql/client.py

```diff
@@ -80,19 +79,30 @@ def format_field(self, value, format_spec):
 
 
 class SPARQLClient:
-    def __init__(self,
-                 endpoint: Optional[str] = None,
+    def __init__(self, endpoint: str, *,
                  update_endpoint: Optional[str] = None,
-                 prefixes: Optional[List[Namespace]] = None,
-                 graph: Optional[str] = None,
+                 prefixes: Optional[Dict[str, IRI]] = None,
+                 graph: Optional[IRI] = None,
                  **kwargs):
         self._closed = False
-        self.endpoint = endpoint or ENV['MU_SPARQL_ENDPOINT']
-        self.graph = graph or IRI(ENV['MU_APPLICATION_GRAPH'])
-        self.update_endpoint = update_endpoint or self.endpoint
+        self._endpoint = endpoint
+        self._update_endpoint = update_endpoint
+        self._graph = graph
         self.session = aiohttp.ClientSession(**kwargs)
         self._generate_prefixes(prefixes)
 
+    @property
+    def endpoint(self):
+        return self._endpoint
+
+    @property
+    def update_endpoint(self):
```

Comment:
hmm... I never thought about that actually. I checked online to see how people call that and apparently the "SPARQL update endpoint" is valid.

https://etl.linkedpipes.com/components/l-sparqlendpoint.html